client/src/tpu_client.rs

```diff
@@ -163,139 +129,18 @@ impl TpuClient {
-                {
-                    if let Ok(result) = self
-                        .rpc_client
-                        .get_signature_statuses(pending_signatures_chunk)
-                    {
-                        let statuses = result.value;
-                        for (signature, status) in
-                            pending_signatures_chunk.iter().zip(statuses.into_iter())
-                        {
-                            if let Some(status) = status {
-                                if status.satisfies_commitment(self.rpc_client.commitment()) {
-                                    if let Some((i, _)) = pending_transactions.remove(signature) {
-                                        confirmed_transactions += 1;
-                                        if status.err.is_some() {
-                                            progress_bar.println(format!(
-                                                "Failed transaction: {:?}",
-                                                status
-                                            ));
-                                        }
-                                        transaction_errors[i] = status.err;
-                                    }
-                                }
-                            }
-                        }
-                    }
-                    spinner::set_message_for_confirmed_transactions(
-                        &progress_bar,
-                        confirmed_transactions,
-                        total_transactions,
-                        Some(block_height),
-                        last_valid_block_height,
-                        "Checking transaction status...",
-                    );
-                }
-
-                if pending_transactions.is_empty() {
-                    return Ok(transaction_errors);
-                }
-            }
-
-            transactions = pending_transactions.into_iter().map(|(_k, v)| v).collect();
-            progress_bar.println(format!(
-                "Blockhash expired. {} retries remaining",
-                expired_blockhash_retries
-            ));
-            expired_blockhash_retries -= 1;
-        }
-        Err(TpuSenderError::Custom("Max retries exceeded".into()))
+        let _guard = RUNTIME.enter();
+        let send_and_confirm_messages_with_spinner = self
+            .tpu_client
+            .send_and_confirm_messages_with_spinner(messages, signers);
+        RUNTIME.block_on(send_and_confirm_messages_with_spinner)
     }
 
     pub fn rpc_client(&self) -> &RpcClient {
         &self.rpc_client
     }
 }
 
-impl Drop for TpuClient {
-    fn drop(&mut self) {
-        self.exit.store(true, Ordering::Relaxed);
-        self.leader_tpu_service.join();
-    }
-}
-
 pub(crate) struct LeaderTpuCacheUpdateInfo {
     pub(crate) maybe_cluster_nodes: Option<ClientResult<Vec<RpcContactInfo>>>,
     pub(crate) maybe_epoch_info: Option<ClientResult<EpochInfo>>,
```

Comment:
This struct only exists for the TPU client, so it can be moved into `nonblocking/tpu_client` and made private once again